### PR TITLE
Docker support

### LINKED ISSUE #1: Provide docker container
Hi, this looks quite nice and so my question would be if it is possible to provide a docker container for this? I've never worked with perl before an honestly don't plan to set it up onto my server bare metal (haven't tried it yet, but definitely keen on trying it), but I think a docker container would make it easier to set up and also easy to daemonize it.

Maybe I'll find some time to contribute docker support myself, but I can't promise that.

### PR DESCRIPTION
**Pull request by cromefire**
Did a quick draft for docker support, but it yet to be tested.

Fixes #1

### CODE REVIEW

**Comment by cromefire** [comment]
Pipeline builds for `amd64`, `armhf` and `aarch64`. Should support most platforms besides really old RPis, but those are a pain to support since many dropped support for arm v6.

**Comment by cromefire** [comment]
Also if you tag a tag like `v1.0.0`, it'll build an the images with the `latest`, `1`, `1.0` and `1.0.0` tag so you might wanna do that after merge.

**Comment by cromefire** [comment]
```
acmeproxy-1  | Configuration file "/app/acmeproxy.pl.conf" missing, maybe you need to create it?
acmeproxy-1 exited with code 0
```
Well it's searching in the wrong folder, no idea why.

**Comment by MadCamel** [comment]
Awesome, thank you!

I think I spotted and fixed the problem. Try re-basing this PR on the latest main and see if it works.

**Comment by cromefire** [comment]
It now has an issue that cron is missing of course, I'll think of something.

**Comment by cromefire** [comment]
To I think it's working now. I have some issues with ZeroSSL though (it's doesn't accept my `+` email, because their email validator seems just bad...) and there's no way currently to pass any options to ACME.sh or so, so I can't specify the `--server` flag to go to letsencrypt.

Not to mention that my configuration would currently reject ZeroSSL certificates anyway because I have `CAA` set up.

**Comment by MadCamel** [comment]
> To I think it's working now. I have some issues with ZeroSSL though (it's doesn't accept my `+` email, because their email validator seems just bad...) and there's no way currently to pass any options to ACME.sh or so, so I can't specify the `--server` flag to go to letsencrypt.

acmeproxy.pl generates a certificate for itself using default acme.sh parameters. You can use acme.sh to provide your own certificate in $HOME/.acme.sh/acmeproxy.pl.key and $HOME/.acme.sh/acmeproxy.pl.crt . It will not overwrite a pre-existing certificate and will even re-load the certificate when it detects the files are modified. Thanks for showing this is required information, I'll see about adding this to the README.

And of course you can pass whatever parameters to acme.sh clients using the proxy without issue.

Is this PR in "works for me" state?

**Comment by cromefire** [comment]
> acmeproxy.pl generates a certificate for itself using default acme.sh parameters. You can use acme.sh to provide your own certificate in $HOME/.acme.sh/acmeproxy.pl.key and $HOME/.acme.sh/acmeproxy.pl.crt .

Yeah I think this isn't on the right volume yet, I might have to change the home folder or so, because I've tried that and it seems it tried to fetch it again after recreating the container.

> Is this PR in "works for me" state?

Almost

**Review by cromefire — COMMENTED** [review]
This has to be addressed before merge.

File: .github/workflows/docker-publish.yml [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+
+# This workflow uses actions that are not certified by GitHub.
+# They are provided by a third-party and are governed by
+# separate terms of service, privacy policy, and support
+# documentation.
+
+on:
+  push:
+    # TODO: Change back to main
+    branches: [ "docker" ]
```
Comment (cromefire) [inline .github/workflows/docker-publish.yml:11]: TODO: This has to be addressed before merge.

Comment (MadCamel) [reply .github/workflows/docker-publish.yml:11]: Awesome. Thank you so much for putting in the work. I'll merge this.

**Pull request merged**